packages/jest-config/src/resolveConfigPath.ts

```diff
@@ -60,15 +60,17 @@ const resolveConfigPathByTraversing = (
   const jestConfig = JEST_CONFIG_EXT_ORDER.map(ext =>
     path.resolve(pathToResolve, getConfigFilename(ext)),
   ).find(isFile);
```

Comment:
this should `filter` and error if the array has a length of more than 1 (i.e. if you have both `jest.config.js` and `jest.config.cjs` we should warn in the same way)

Reply:
Oh, ok. Misunderstood the issue here.

This might lead to little refactor as the order of extensions doesn't matter anymore (which is mentioned in multiple places).

Reply:
This one is finished with a unit test. I also had to change the wording (since there could be potentially more configs).